src/geotiffwriter.js

```diff
@@ -251,17 +252,47 @@ const encodeImage = (values, width, height, metadata) => {
   }
 
   const prfx = new Uint8Array(encodeIfds([ifd]));
+  const samplesPerPixel = ifd[277];
 
-  const img = new Uint8Array(values);
+  const dataType = values.constructor.name;
+  const TypedArray = typeMap[dataType];
 
-  const samplesPerPixel = ifd[277];
+  let elementSize = 4;
```

Comment:
could you add a comment describing why you set the default to 4?

Reply:
could we consider defaulting elementSize to 8, so we can support all floating point numbers by default?